### PR TITLE
Align CordaRPCOps Vault Query API's for Java/Kotlin with VaultQuery s…

### PR DESCRIPTION
**Pull request by josecoll**
…ervice equivalents.

Added Kotlin extension functions for ease of use in simplest case.
Added missing Java contractType class (previously was forced cast of ContractState::class)

### CODE REVIEW

File: client/rpc/src/smoke-test/kotlin/net/corda/kotlin/rpc/StandaloneCordaRPClientTest.kt [hunk 1]
```diff
@@ -11,10 +11,7 @@ import net.corda.core.contracts.SWISS_FRANCS
 import net.corda.core.crypto.SecureHash
 import net.corda.core.getOrThrow
 import net.corda.core.identity.Party
-import net.corda.core.messaging.CordaRPCOps
```
Comment (shamsasari) [inline client/rpc/src/smoke-test/kotlin/net/corda/kotlin/rpc/StandaloneCordaRPClientTest.kt]: Unrelated changes onto an otherwise untouched file.

File: core/src/main/kotlin/net/corda/core/messaging/CordaRPCOps.kt [hunk 2]
```diff
@@ -104,20 +106,23 @@ interface CordaRPCOps : RPCOps {
...
 
     // DOCSTART VaultQueryAPIJavaHelpers
-    fun <T : ContractState> vaultQueryByCriteria(criteria: QueryCriteria): Vault.Page<T> = vaultQueryBy(criteria)
-    fun <T : ContractState> vaultQueryByWithPagingSpec(criteria: QueryCriteria, paging: PageSpecification): Vault.Page<T> = vaultQueryBy(criteria, paging)
-    fun <T : ContractState> vaultQueryByWithSorting(criteria: QueryCriteria, sorting: Sort): Vault.Page<T> = vaultQueryBy(criteria, sorting = sorting)
-
-    fun <T : ContractState> vaultTrackByCriteria(criteria: QueryCriteria): Vault.PageAndUpdates<T> = vaultTrackBy(criteria)
-    fun <T : ContractState> vaultTrackByWithPagingSpec(criteria: QueryCriteria, paging: PageSpecification): Vault.PageAndUpdates<T> = vaultTrackBy(criteria, paging)
-    fun <T : ContractState> vaultTrackByWithSorting(criteria: QueryCriteria, sorting: Sort): Vault.PageAndUpdates<T> = vaultTrackBy(criteria, sorting = sorting)
+    fun <T : ContractState> vaultQuery(contractType: Class<out T>): Vault.Page<T> = vaultQueryBy(QueryCriteria.VaultQueryCriteria(), contractType = contractType)
```
Comment (shamsasari) [inline core/src/main/kotlin/net/corda/core/messaging/CordaRPCOps.kt]: Can you convert these to have code blocks to avoid the long lines.

File: core/src/main/kotlin/net/corda/core/messaging/CordaRPCOps.kt [hunk 3]
```diff
@@ -104,20 +106,23 @@ interface CordaRPCOps : RPCOps {
     @RPCReturnsObservables
     fun <T : ContractState> vaultTrackBy(criteria: QueryCriteria = QueryCriteria.VaultQueryCriteria(),
                                          paging: PageSpecification = PageSpecification(),
-                                         sorting: Sort = Sort(emptySet())): Vault.PageAndUpdates<T>
+                                         sorting: Sort = Sort(emptySet()),
+                                         contractType: Class<out T>): Vault.PageAndUpdates<T>
```
Comment (shamsasari) [inline core/src/main/kotlin/net/corda/core/messaging/CordaRPCOps.kt]: The positioning of `contractType` here is inconsistent with the positioning in the new methods below.

File: core/src/main/kotlin/net/corda/core/messaging/CordaRPCOps.kt [hunk 4]
```diff
@@ -104,20 +106,23 @@ interface CordaRPCOps : RPCOps {
     @RPCReturnsObservables
     fun <T : ContractState> vaultTrackBy(criteria: QueryCriteria = QueryCriteria.VaultQueryCriteria(),
```
Comment (shamsasari) [inline core/src/main/kotlin/net/corda/core/messaging/CordaRPCOps.kt]: For consistency between Kotlin and Java, should we be providing default values here now that we have new methods below?

Comment (josecoll) [inline core/src/main/kotlin/net/corda/core/messaging/CordaRPCOps.kt]: I can change them - but then should also change the ones in VaultQueryService

File: client/rpc/src/smoke-test/kotlin/net/corda/kotlin/rpc/StandaloneCordaRPClientTest.kt (hunk 1 above)
Comment (josecoll) [inline client/rpc/src/smoke-test/kotlin/net/corda/kotlin/rpc/StandaloneCordaRPClientTest.kt]: Not sure I follow.
Reorganise imports changed this:
import net.corda.core.messaging.*

Comment (shamsasari) [inline client/rpc/src/smoke-test/kotlin/net/corda/kotlin/rpc/StandaloneCordaRPClientTest.kt]: The file doesn't have any other change besides this, so it's a cosmetic change. But this is a minor issue.

File: core/src/main/kotlin/net/corda/core/messaging/CordaRPCOps.kt [hunk 5]
```diff
@@ -102,22 +104,41 @@ interface CordaRPCOps : RPCOps {
...
+    fun <T : ContractState> vaultTrackBy(criteria: QueryCriteria,
+                                         paging: PageSpecification,
+                                         sorting: Sort,
+                                         contractType: Class<out T>): Vault.PageAndUpdates<T>
     // DOCEND VaultTrackByAPI
 
     // Note: cannot apply @JvmOverloads to interfaces nor interface implementations
     // Java Helpers
 
     // DOCSTART VaultQueryAPIJavaHelpers
```
Comment (shamsasari) [inline core/src/main/kotlin/net/corda/core/messaging/CordaRPCOps.kt]: The wording in valut-query.rst needs to slightly change as it says these are Java helpers, which is no longer the case.

File: core/src/main/kotlin/net/corda/core/messaging/CordaRPCOps.kt [hunk 6]
```diff
@@ -102,22 +104,41 @@ interface CordaRPCOps : RPCOps {
...
-    fun <T : ContractState> vaultQueryByCriteria(criteria: QueryCriteria): Vault.Page<T> = vaultQueryBy(criteria)
-    fun <T : ContractState> vaultQueryByWithPagingSpec(criteria: QueryCriteria, paging: PageSpecification): Vault.Page<T> = vaultQueryBy(criteria, paging)
-    fun <T : ContractState> vaultQueryByWithSorting(criteria: QueryCriteria, sorting: Sort): Vault.Page<T> = vaultQueryBy(criteria, sorting = sorting)
+    fun <T : ContractState> vaultQuery(contractType: Class<out T>): Vault.Page<T> {
```
Comment (shamsasari) [inline core/src/main/kotlin/net/corda/core/messaging/CordaRPCOps.kt]: Perhaps move these methods below vaultQueryBy, and likewise with vaultTrackBy
